src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,572 @@
+  SetInputParam("min", minVal);
+  SetInputParam("max", maxVal);
+  SetInputParam("distances_file", distanceFile);
+  SetInputParam("neighbors_file", neighborsFile);
+
+  mlpackMain();
+
+  SetInputParam("input_model", move(CLI::GetParam<RSModel*>("output_model")));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  remove(neighborsFile.c_str( ));
+  remove(distanceFile.c_str( ));
+}
+
+/*
+ * Check that we cannot specify a tree type which is not available or wrong
+ */
+BOOST_AUTO_TEST_CASE(RangeSearchDifferentTree)
+{
+  arma::mat inputData;
+  double minVal = 0, maxVal = 3;
+  string distanceFile = "distances.csv";
+  string neighborsFile = "neighbors.csv";
+  string wrongTreeType = "RST";
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("reference", move(inputData));
+  SetInputParam("min", minVal);
+  SetInputParam("max", maxVal);
+  SetInputParam("distances_file", distanceFile);
+  SetInputParam("neighbors_file", neighborsFile);
+  SetInputParam("tree_type", wrongTreeType);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  remove(neighborsFile.c_str( ));
+  remove(distanceFile.c_str( ));
+}
+
+/*
+ * Check that we cannot specify both a Reference and Input Model
+ */
+BOOST_AUTO_TEST_CASE(RangeSearchBothReferenceandModel)
+{
+  arma::mat inputData, queryData;
+  double minVal = 0, maxVal = 3;
+  string distanceFile = "distances.csv";
+  string neighborsFile = "neighbors.csv";
+
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+  if (!data::Load("iris_test.csv", queryData))
+    BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+  SetInputParam("reference", move(inputData));
+  SetInputParam("min", minVal);
+  SetInputParam("max", maxVal);
+  SetInputParam("distances_file", distanceFile);
+  SetInputParam("neighbors_file", neighborsFile);
+  SetInputParam("query", queryData);
+
+  mlpackMain();
+
+  SetInputParam("input_model", move(CLI::GetParam<RSModel*>("output_model")));
+  SetInputParam("query", move(queryData));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  remove(neighborsFile.c_str( ));
+  remove(distanceFile.c_str( ));
+}
+
+/*
+* Check that the correct output is returned for a small synthetic input
+* case , where the parameters of location , min value and max value are provided
+* and are checked with pre-calculated neighbor and distance values
+*/
+BOOST_AUTO_TEST_CASE(RangeSearchTest)
+{
+  //Matrix Input is expected in this format
+  arma::mat x = {{0, 3, 3, 4, 3, 1},
+                 {4, 4, 4, 5, 5, 2},
+                 {0, 1, 2, 2, 3, 3}};
+  std::string distanceFile = "distances.csv";
+  std::string neighborsFile = "neighbors.csv";
```

Comment:
Since you have `using namespace std` you can actually just call this `string` not `std::string`. :)